python/triton/language/libdevice.py

```diff
@@ -0,0 +1,1126 @@
+from . import core, extern
+
+
+@extern.extern
+def clz(arg0, _builder=None):
+    return extern.elementwise("libdevice", "/usr/local/cuda/nvvm/libdevice/libdevice.10.bc", [arg0, ], {(core.dtype("int32"),): ("__nv_clz", core.dtype("int32")), (core.dtype("int64"),): ("__nv_clzll", core.dtype("int32")), }, _builder)
```

Comment:
@ptillet Do you want to replace `/usr/local/cuda/nvvm/libdevice/libdevice.10.bc` with `os.environ['HOME']`?

Reply:
There's also a more flexible calling convention as illustrated in the tutorial:
```
kernel[(1,)](x_tri,` y_tri, BLOCK=shape[0], extern_libs={'libdevice': lib_path})
```

Reply:
Do you think `/usr/local/cuda/nvvm/libdevice/libdevice.10.bc` could be moved into a variable at the beginning of the file, i.e. `LIBDEVICE_PATH=/usr/local/cuda/nvvm/libdevice/libdevice.10.bc` ? Let's not bother with an additional environment variable yet unless it proves useful in the future. The `extern_libs` dict is cool, especially for user-defined external libraries.

Reply:
Done, thanks for the suggestion